SignalMessaging/ViewControllers/OWSNavigationController.m

```diff
@@ -107,12 +107,22 @@ - (BOOL)gestureRecognizerShouldBegin:(UIGestureRecognizer *)gestureRecognizer
     }
 }
 
-
 #pragma mark - NavBarLayoutDelegate
 
 - (void)navBarCallLayoutDidChangeWithNavbar:(OWSNavigationBar *)navbar
 {
     [self updateLayoutForNavbar:navbar];
+    [self setNeedsStatusBarAppearanceUpdate];
+}
+
+- (UIStatusBarStyle)preferredStatusBarStyle
+{
+    if (OWSWindowManager.sharedManager.hasCall) {
+        // Status bar is overlaying the green "call banner"
+        return UIStatusBarStyleLightContent;
+    } else {
+        return super.preferredStatusBarStyle;
+    }
```

Comment:
Will this do the right thing during screen lock/protection?

Reply:
Also, let's verify the registration flow. And the upgrade splashes.

Reply:
Screen lock is above the status bar, good call on the upgrade splashes and registration flow. 

This change works for them as well, but I neglected to update their header style - I'll do that now.